### DISCUSSION TITLE
(Exit with code: 1 and signal: null ) bu ve benzer

### DISCUSSION THREAD

**Discussion by hatqi** [19/classroom#35] [Anonim] [answered]
![image](https://user-images.githubusercontent.com/100834264/158796341-1be5475b-1f47-47bf-9882-5df9af967186.png)
**Ekran Alıntısı 1**

Ekran Alıntısı 1'de görüldüğü üzere basit bir program denedim. Karşılaştığım sonuç ise Ekran Alıntısı 2 de görülmekte. Denediğim program test aşamasında başarısız oluyor. Ve test sonucu da Ekran Alıntısı 2'de görülüyor.

![image](https://user-images.githubusercontent.com/100834264/158796799-86c9ef66-e993-4c1d-ae06-15f08eed0dc1.png)
**Ekran Alıntısı 2**

Sonra benzer sonucu bir başka program üzerinde yaşadım. Bu programın kodları ise Ekran Alıntısı 3'de görülüyor. 

![image](https://user-images.githubusercontent.com/100834264/158799488-e03de5a8-5bfd-458d-9290-da9df0427ef3.png)
**Ekran Alıntısı 2**

Otomatik test sonucu Ekran Alıntısı 3'de gözüküyor.

![image](https://user-images.githubusercontent.com/100834264/158800491-1f91e893-2cbb-4232-ab98-6b09d50dafdd.png)
**Ekran Alıntısı 3**

Ekran Alıntısı 4'te ise sorunsuz çalışan ve testlerden geçen yazılımın bir parçası var.
![image](https://user-images.githubusercontent.com/100834264/158801700-99f4b74f-a5fe-46d2-9085-6f6315b1af54.png)
**Ekran Alıntısı 4**

Sorunsuz çalışan programın sadece puts komutunda yaptığım basit bir değişiklik ise test aşamasının olumsuz sonuçlanmasına sebep oluyor. 
![image](https://user-images.githubusercontent.com/100834264/158802156-b6c8ee8f-c815-44f2-85f2-a78c73d14e62.png)
**Ekran Alıntısı 5**

Bu sorunların kaynağı nedir ve nasıl çözeriz? Gözden kaçan bir durum var sanırım ama nedir?
Şimdiden Teşekkürler.

**Comment by roktas** [comment, marked as answer]
@hatqi İşleri çok karıştırmışsın.  Önce temel bir konuyu anlaman gerekiyor.  Her ödevin ayrı bir test seti var ve bu test setini biz sadece o ödeve özel olarak depoya ekliyoruz.  Yani şöyle bir şey yok: "Şuraya bir kod yazayım göndereyim, Github'ta doğru olup olmadığını söylesin".  Özetle, kodu gönderdiğinde bunun istenen doğru kod olup olmadığını anlayan bir yapay zeka sistemi yok karşı tarafta.  Böyle bir yapay zeka geliştiremezsin de ayrıca.  Programın doğruluğunu/yanlışlığını bir şekilde tanımlamak zorundasın sisteme değil mi?  Senin zihninde geçenleri okuyacak bir sistem mi bekliyorsun?  İşte ödev testleri buna yarıyor.  "Doğruluk" dediğimiz şeyi tanımlıyor.

Somut olarak anlatacak olursak:

- Bu ödevle hiç bir ilişkisi olmayan bir kodu, benim Youtube videosunda verdiğim bir örneği, ödevdeki koda yapıştırmışsın.  Testler de hata veriyor doğal olarak.  (Yukarıda yaptığım açıklamayı dikkate aldığında) aksini beklemek tuhaf olmaz mı?

- Ekran alıntısı 5'te çıktıya `Sonuç=` gibi bir şey eklemişsin.  Bu senin için anlamlı, anlaşılır ve doğru olabilir ama testtler için durum öyle değil.  Programın doğruluğu basitçe ekranda `5.0` çıktısının alınması.  İlave bir şey eklersen testten de geçemezsin değil mi?  `Sonuç=5.0` ile `5.0` aynı metinler mi?

Diğer yazdıklarına madde madde girmeyeyim artık, temel noktaları belirttim.  Fakat bir kaç konuda uyarı yapayım sana:

- Kod yapıştırmayın dedim hala kod yapıştırıyorsun.  Sorununu anlatmanın tek yolu bu olduğundan böyle yaptın ama sana duyurduğum bir kuralı da ihlal etmiş oldun (kuralı bile göz ardı etmenin nedeni son maddede bahsettiğim hususla ilgili değildir umarım).  Tekrarı olmasın...

- `.` ile başlayan dosya ve dizinlere müdahale etmeyin dedik ama bakıyorum depoda `.github/workflow/ruby.yml` dosyası var, yani müdahale etmeyin dediğim yere dosya eklemişsin.  Bunu yanlışlıkla yaptığın belli ama bir daha tekrar etmesin.  Kurallara mutlaka uyun.

- Son olarak...  "Ödevden çok ödev sistemiyle uğraşma madalyası" sana gidiyor:  :1st_place_medal:  Zaten tamamlamış ve tam puan almış olduğun bir ödevde bu kadar uğraşmanı biraz hırs yaptın gibi mi yorumlasam bilemedim?  :-)  Aman sakin...

**Reply by hatqi** [reply]
Madalya için teşekkür ederim. :) Hırs değil, yapmış olduğum işi anlama meselesi bir nevi **Neden?** sorusunca cevap aramak diyebilirim. Çünkü "**Önce temel bir konuyu anlaman gerekiyor. Her ödevin ayrı bir test seti var ve bu test setini biz sadece o ödeve özel olarak depoya ekliyoruz.** " bu konuyu bilmiyordum şimdi öğrendim ve neden sorusunun cevabına ulaştım diyebilirim. Ancak hala kafama yatmayan konular var. Süreç içerisinde cevabını bulacağımı düşünüyorum. İyi çalışmalar.

**Comment by roktas** [comment]
Neymiş kafana yatmayan?

**Reply by hatqi** [reply]
Boşluklar. Boşluklar da amı sizin test setinizin bir kuralı mı?

**Comment by roktas** [comment]
@hatqi Evet öyle, hangi boşluk olduğuna, nerede kullanıldığına göre bazı kurallar tanımlı.  Bu hafta derste hatırlat bana, buraya uzun yazmadan anlatayım.  Bu arada aferin, ödevi başarıyla teslim eden ilk bir kaç kişiden birisin.

**Reply by hatqi** [reply]
👍 Teşekkürler.
